b2sdk/filter.py

```diff
@@ -0,0 +1,53 @@
+######################################################################
+#
+# File: b2sdk/filter.py
+#
+# Copyright 2024 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+from __future__ import annotations
+
+import fnmatch
+from enum import Enum
+from typing import NamedTuple, Sequence
+
+
+class FilterType(Enum):
+    INCLUDE = "include"
+    EXCLUDE = "exclude"
+
+
+class Filter(NamedTuple):
+    type: FilterType
+    pattern: str
```

Comment:
If I'm not mistaken, this is the first use of `NamedTuple` in b2sdk's public interface. Is there a particular reason for `NamedTuple` here? If not, I'd probably go for a `dataclass` just for consistency.